### DISCUSSION TITLE
Should we release 0.72.6?

### DISCUSSION THREAD

**Discussion by Titozzz** [reactwg/react-native-releases#89] [Patches]
## Released, please refer to #91

Current Release: [0.72.5](https://github.com/facebook/react-native/releases/tag/v0.72.5)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### Requests that didn't make it in 0.72.5

- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-6721723
- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-6975730
- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-6975769
- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-6975987
- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-6990958
- https://github.com/reactwg/react-native-releases/discussions/85#discussioncomment-7077577

#### List of qualified picks

- [x] https://github.com/facebook/react-native/pull/39575 
- [x] https://github.com/facebook/react-native/commit/63884b48882f737fa652e07227800c9baea6197c Update Xcode 15 patches to be more robust
- [x] https://github.com/facebook/react-native/commit/ce39931bc2b02f13cbc5751ba4d4a6dbc07bc91a Fix Gemfile, setting Active support to < 7.1.0
- [x] https://github.com/facebook/react-native/commit/8b768f144a3cad25d84a4e70b880e5aa1d5ff872 RN: Switch EventEmitter to `Array.from(...)`

#### Local commits to backport to main

1.

**Comment by liamjones** [comment]
The last patch release had three fixes for Xcode 15 build issues. I've just noticed Cocoapods released a new version on Friday with an Xcode 15 related fix; https://github.com/CocoaPods/CocoaPods/releases/tag/1.13.0. Should the RN template Gemfile also be specifying this as a minimum version of Cocoapods now or is the fix not relevant to RN?

**Comment by lorenc-tomasz** [comment]
I think 0.72.6 should fix issues with Xcode 14. When someone uses Xcode 15 and push changes generated with `pod install` to projects that adds new flags:

```
OTHER_LDFLAGS = (
	"$(inherited)",
	"-Wl",
	"-ld_classic",
	"-Wl",
	"-ld_classic",
);
```

and probably `_LIBCPP_ENABLE_CXX17_REMOVED_UNARY_BINARY_FUNCTION,`

the changes are not removed when new `pod install` action is done with Xcode 14 (on CI) and the build fails with `ld: library not found for -ld_classic`.

EDIT:

On Azure DevOps new version of macOS-13 image was released with stable Xcode 15 so I can switch to new Xcode.

**Reply by cipolleschi** [reply]
This should be already the case as:
- [this](https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/cocoapods/utils.rb#L164-L171) is the code block that decides whether to add or remove the flags.
- [this](https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/cocoapods/utils.rb#L342-L347) is the function that adds the flag and should add them only if they are not in the list already
- [this](https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/cocoapods/utils.rb#L349-L359) is the function that removes them. And again, it should remove the values. 

We also have [unit tests](https://github.com/facebook/react-native/blob/main/packages/react-native/scripts/cocoapods/__tests__/utils-test.rb#L531-L670) and I tested it manually... I'm very confused about this issue.

I'll try with a different implementation to see if it works better.

**Comment by MO-Lewis** [comment]
It's already in your "Requests that didn't make it in 0.72.5" list, but posting again for good measure:

Would be much appreciated if you could pick this:
https://github.com/facebook/react-native/commit/0cdb9e6a52f69ebee635d5bfaa34a52f6059dd47

It features an important couple of fixes which help to better support SSL Pinning within React Native. It was previously approved some years back, but had to be resubmitted again.

Thanks!

**Comment by cipolleschi** [comment]
Issue #39692 needs to be fixed.

**Reply by huntie** [reply]
Resolved by https://github.com/facebook/react-native/pull/39698.

**Comment by cipolleschi** [comment]
Two picks, after they get merged:
- https://github.com/facebook/react-native/pull/39575 (this is a pr against 0.72-stable already!)
- https://github.com/facebook/react-native/pull/39710

**Reply by lorenc-tomasz** [reply]
Great :) Thank you :)

**Reply by huntie** [reply]
Both changes added to branch ✅

**Comment by cipolleschi** [comment]
Cherry-pick: https://github.com/facebook/react-native/commit/ce39931bc2b02f13cbc5751ba4d4a6dbc07bc91a

They released ActiveSupport 7.1.0 which breaks Cocoapods. 😮‍💨
This should fix the issue for all our users as they should use Bundler.

**Reply by huntie** [reply]
Done

**Comment by Titozzz** [comment]
https://github.com/facebook/react-native/commit/58adc5e4b9ab74b67b4af04d1e72c387af848ea7 was picked in 72.2 and caused a regression fixed by https://github.com/facebook/react-native/commit/16fc065aa1a417126ca008d1de7691258fcb0fad, so we should pick it

**Comment by yungsters** [comment]
Can we pick https://github.com/facebook/react-native/commit/8b768f144a3cad25d84a4e70b880e5aa1d5ff872 for a patch release for v72?

It is a pretty safe fix for a relatively disruptive issue that has affected some folks.

**Reply by huntie** [reply]
Done

**Comment by joe-sam** [comment]
Can we pick 
 **Prevent LogBox from crashing on long messages** (https://github.com/facebook/react-native/pull/38005)

**Comment by stianjensen** [comment]
Regression with rotation in animated values in RN0.72:
https://github.com/facebook/react-native/commit/ae0d714bbdd2eb9d114c17c3945b2f678f60b65a
(explained here: https://github.com/facebook/react-native/issues/36608)
